crates/wasmtime/src/runtime/vm/vmcontext.rs

```diff
@@ -1173,6 +1173,10 @@ pub struct VMStoreContext {
     /// on `VMStackChain` for details.
     pub stack_chain: UnsafeCell<VMStackChain>,
 
+    /// A pointer to the embedder's `T` inside a `Store<T>`, for use with the
+    /// `store-data-address` unsafe intrinsic.
+    pub store_data: Option<VmPtr<()>>,
```

Comment:
To avoid the `Option` here, could this use `NonNull::dangling()` as an initial constructor? That'll still segfault if erroneously accessed but otherwise avoids the pesky `unwrap`s